clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3556,6 +3590,37 @@ void Sema::AddSYCLIntelMaxWorkGroupSizeAttr(Decl *D,
   if (!XDim || !YDim || !ZDim)
     return;
 
+  // If the [[intel::max_work_group_size(X, Y, Z)]] attribute is specified on
+  // a declaration along with [[sycl::reqd_work_group_size(X1, Y1, Z1)]]
+  // attribute, check to see if values of reqd_work_group_size arguments are
+  // equal or less than values of max_work_group_size attribute arguments.
+  //
+  // The arguments to reqd_work_group_size are ordered based on which index
+  // increments the fastest. In OpenCL, the first argument is the index that
+  // increments the fastest, and in SYCL, the last argument is the index that
+  // increments the fastest.
+  //
```

Comment:
`__attribute__((reqd_work_group_size(1, 2, 3)))`` and ``[[cl::reqd_work_group_size(1, 2, 3)]]`` both have been deprecated in favor of `` [[sycl::reqd_work_group_size()]]``  on https://github.com/intel/llvm/pull/3689

``[[intel::reqd_work_group_size(1,2,3)]]`` has been renamed to ``[[sycl::reqd_work_group_size(1,2,3)]]``
OpenCL and SYCL orderings are based on discussion below:
https://github.com/intel/llvm/issues/3252#issuecomment-785261128